image-loader/app/lib/ImageProcessorClassReflector.scala

```diff
@@ -0,0 +1,47 @@
+package lib
+
+import com.gu.mediaservice.lib.cleanup.{ImageProcessor, ImageProcessorConfig}
+import com.gu.mediaservice.lib.config.CommonConfig
+import com.typesafe.scalalogging.StrictLogging
+
+import scala.reflect.runtime.universe
+import scala.util.control.NonFatal
+
+object ImageProcessorClassReflector extends StrictLogging {
+  def catchNonFatal[T](block: => T): Either[Throwable, T] = {
+    try {
+      Right(block)
+    } catch {
+      case NonFatal(e) => Left(e)
+    }
+  }
+
+  def loadImageProcessor(className: String, config: ImageProcessorConfig): Either[String, ImageProcessor] = {
+    val mirror = universe.runtimeMirror(getClass.getClassLoader)
```

Comment:
Oh wow, I had forgotten how verbose Scala reflection is. Lets just use the Java API instead: https://docs.oracle.com/javase/tutorial/reflect/member/ctorInstance.html

Reply:
Converted and basic unit tests added.